src/section/mod.rs

```diff
@@ -0,0 +1,200 @@
+/// every line.
+///
+/// # Construction
+///
+/// Sections are meant to be constructed via `Into<Section>`, which is implemented for all types
+/// that implement `Display`. The constructed `Section` then takes ownership it the `Display` type
+/// and boxes it internally for use later when printing the report.
+///
+/// # Examples
+///
+/// ```rust
+/// use color_eyre::{SectionExt, Help, Report};
+/// use eyre::eyre;
+/// use std::process::Command;
+/// use tracing::instrument;
+///
+/// trait Output {
+///     fn output2(&mut self) -> Result<String, Report>;
+/// }
+///
+/// impl Output for Command {
+///     #[instrument]
+///     fn output2(&mut self) -> Result<String, Report> {
+///         let output = self.output()?;
+///
+///         let stdout = String::from_utf8_lossy(&output.stdout);
+///
+///         if !output.status.success() {
+///             let stderr = String::from_utf8_lossy(&output.stderr);
+///             Err(eyre!("cmd exited with non-zero status code"))
+///                 .with_section(move || {
+///                     "Stdout:"
+///                         .skip_if(|| stdout.is_empty())
+///                         .body(stdout.trim().to_string())
+///                 })
+///                 .with_section(move || {
+///                     "Stderr:"
+///                         .skip_if(|| stderr.is_empty())
+///                         .body(stderr.trim().to_string())
+///                 })
+///         } else {
+///             Ok(stdout.into())
+///         }
+///     }
+/// }
+/// ```
+pub struct Section {
+    pub(crate) header: Box<dyn Display + Send + Sync + 'static>,
+    pub(crate) body: Option<Box<dyn Display + Send + Sync + 'static>>,
+    pub(crate) order: Order,
+}
+
+/// Extension trait for customizing the content of a `Section`
+pub trait SectionExt {
+    /// Add a body to a `Section`
+    ///
+    /// Bodies are always indented to the same level that error messages and spans are indented.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use color_eyre::{Help, SectionExt, Report};
+    /// use eyre::eyre;
+    ///
+    /// let all_in_header = "header\n   body\n   body";
+    /// let report = Err::<(), Report>(eyre!("an error occurred"))
+    ///     .section(all_in_header)
+    ///     .unwrap_err();
+    ///
+    /// let just_header = "header";
+    /// let just_body = "body\nbody";
+    /// let report2 = Err::<(), Report>(eyre!("an error occurred"))
+    ///     .section(just_header.body(just_body))
+    ///     .unwrap_err();
+    ///
+    /// assert_eq!(format!("{:?}", report), format!("{:?}", report2))
+    /// ```
+    fn body<C>(self, body: C) -> Section
+    where
+        C: Display + Send + Sync + 'static;
+
+    /// Skip printing `Section` if condition is true
+    ///
+    /// Useful for skipping sections based on the content of its body, even if the section header
+    /// is not empty
```

Comment:
A prose example might be more useful here, and breaking out the condition as a separate sentence.

> Skip a section based on the content of its body. For example, skip a section if the body is empty.
>
> If this condition is true, the section will be skipped even if the header is not empty.